### ISSUE TITLE
(Docs) AJAX Rendering formats

### ISSUE THREAD

**Issue by PoDuck** [fooplugins/FooTable#575]
I am creating my tables using JSON data provided by the server.  The problem I have is that I can't figure out how one sets things like "data-sort-value" on the individual <td> elements dynamically.  Looking around the docs, I don't see anything referencing this, nor do I see examples of this, and considering that I am less than a novice when it comes to javascript, I am flummoxed reading the code to figure it out.

I figured out how to set the values in the <th> elements, but only through guesses based on what I saw in the columns.json files available.  I don't see such examples for setting the <td> elements, and I frankly don't even know if it is an available option.  If it isn't, it should be.  If it is, it should be documented.

Considering that ajax loading is a very useful component of footable, I would think that at least a passing reference to these things should be easily found in the docs.  Maybe a section about passing attributes through JSON should be added, for both the columns and the rows.  In the mean time, could someone enlighten me?

**Comment by PoDuck** [comment]
Okay, after much searching, I think I figured it out.  It is in the documentation.  It's just not as easy to find as I thought it would be.

Sorry.

**Comment by jeffconklin** [comment]
Can you include a link / some direction to what you found for future reference here? Anyone else coming here would appreciate it.

**Comment by PoDuck** [comment]
I thought I found it under Getting Started in the cell options, but that still isn't it, so I am reopening this question.

It is kind of misleading in the docs actually.  It's showing:
```
<table>
    ..
        <tr>
            <td data-value="1"><!-- the formatter result --></td>
                ...
          </tr>
    ...
</table>
```

While `data-sort-value` is what I am trying to set on individual cells, I figured that if I put `sortValue` there, it might add a `data-sort-value` attribute, but when I use either set of code below it, it only inserts a value between the `<td>` and `</td>` tags.  It doesn't set the data-value.  I am trying to be able to sort tags with symbols in them, so the only way I can do it is to set a value for sorting.

This is one of the sets of code it gives for the javascript:
```
jQuery(function($){
    $('.table').footable({
        "rows": [{
            "id": 1
        },{
            ...
        }]
    });
});
```

Here is something similar to some code I tried:
```
jQuery(function($){
    $('.table').footable({
        "rows": [{
            "id": {
                "options": {
                    "sortVallue": 1
                },
                "value": 1
            }
        },{
            ...
        }]
    });
});
```
I have also tried permutations with the "value" attribute as well, and it does the same thing.

**Comment by PoDuck** [comment]
I also want to add that I have the data-type of the column set to html, which is said to be what the sort-value needs.

**Comment by PoDuck** [comment]
I'm not seeing the problem.  Maybe someone else can tell me if I'm doing something wrong.

My columns JSON is [here](http://127.0.0.1:8000/api/employees/columns/), and my rows are [here](http://127.0.0.1:8000/api/employees/).  Not sure that the columns matter, but tossed them in anyway.

**Comment by jeffconklin** [comment]
I re-read your questions think I understand what you are looking for. You are trying to include both a sort value and the "visual" value for the cell, right? That is an option, and is pretty much what you already shared above where you set the value and set options->sortValue...

`jQuery(function($){
    $('.table').footable({
        "rows": [{
            "id": {
                "options": {
                    "sortVallue": "Sorted Value"
                },
                "value": "Visual Value"
            }
        },{
            ...
        }]
    });
});`

I created a JSBin snippet with an example here: https://jsbin.com/simepog/edit    I split up first entry within "rows" to be easier to read, including my "title" column:

`"title" : {
	"options" : {"sortValue" : "18 Hole Disc Golf Course at Ocean County Park"},
	"value" : "<a href=\"#\">18 Hole Disc Golf Course at Ocean County Park</a> "
},`

This allows a link to be displayed in the cell, and the text of "18 Hole Disc Golf Course at Ocean County Park" to be used with sorting.

Can you create a Fiddle or something similar with what you are trying to do / some data?

Also wanted to mention the JSON links you provided goes to a local 127.0.0.1 address and are not accessible.

_**Edited to change example code**_

**Comment by PoDuck** [comment]
Thank you for your help.  Your example was exactly what I was looking for.

I just realized that those links I shared were local links for the rows and columns, so they won't work. but I finally got it working, and you did explain it correctly, although slightly too late to help me, as I figured it out a few hours ago.

I actually think I was doing it correctly with my example above, but it wasn't working at the time for some reason.  I reworked my code, and now it works great, although I can't figure out what I did differently.

Anyway, and I don't mean to sound nit picky, but it might be helpful in the docs if someone were to include a little extra tidbit on how to format the JSON in general, and not just for each individual item.  I was originally under the assumption that the individual cells could not have any attributes set on them through JSON, since I didn't see examples right off.  It appears that it is all documented, for instance when going to the sort link from the top menu, each individual command is documented with examples in javascript, but a generic *for everything* document that can be found easily, maybe on the front page, or the ajax page, would have gotten me there more quickly.
